File: 1482-minimum-number-of-days-to-make-m-bouquets/1482-minimum-number-of-days-to-make-m-bouquets.py
class Solution:
    def minDays(self, bloomDay: List[int], m: int, k: int) -> int:
        n = len(bloomDay)
        if(n < m * k):
            return -1

        low = 1
        high = max(bloomDay)

        def canMake(mid):
            made = 0
            f = 0
            for i in range(n):
                if(bloomDay[i] <= mid):
                    f += 1
                else:
                    f = 0 
                if(f == k):
                    made += 1
                    f = 0

            return made >= m


        res = 0
        while(low <= high):
            mid = (high -  low) // 2 + low
            if(canMake(mid)):
                high = mid - 1
                res = mid
            else:
                low = mid + 1

        return res

File: 1482-minimum-number-of-days-to-make-m-bouquets/1482-minimum-number-of-days-to-make-m-bouquets.go
func minDays(bloomDay []int, m int, k int) int {
    var n = len(bloomDay)
    if(n < m * k) { return -1 }
    
    var getKCounts = func (days int) int {
        var consCount, kCons = 0, 0
        for _, b := range bloomDay {
            if consCount == k {
                kCons++
                consCount = 0
            }
            if b > days{
                consCount = 0
            } else {
                consCount++
            }
        }
        
        return kCons
    }
    
    
    var low, high = 1, 0
    for _, b := range bloomDay {
        high = max(high, b)
    }
    
    var res = -1
    bloomDay = append(bloomDay, high + 1)
    for low <= high {
        var mid = (low + high) >> 1
        var kc = getKCounts(mid)
        if kc >= m {
            res = mid
            high = mid - 1
        } else {
            low = mid + 1
        }
    }
    
    return res
}